src/SDKs/ContainerInstance/ContainerInstance.Tests/ContainerGroupScenarioTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Azure.Management.ContainerInstance;
+using Microsoft.Azure.Management.ContainerInstance.Models;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Rest.Azure;
+using Xunit;
+
+namespace ContainerInstance.Tests
+{
+	public class ContainerGroupScenarioTests
+	{ 
+		[Fact]
+		public void ContainerGroupCreateGetDeleteTests()
+		{
+			using (TestContext context = new TestContext(this))
+			{
+				string containerInstanceLocation = "westus";
+
+				ResourceGroup resourceGroup = context.CreateResourceGroup("containergroupcrd-", containerInstanceLocation);
+				ContainerInstanceManagementClient containerClient = context.GetClient<ContainerInstanceManagementClient>();
+
+				string containerGroupName = context.GenerateName("containergroup");
+				string containerOsType = "Linux";
+
+				string containerName = "test1";
+				string containerImage = "nginx";
+				int containerPort = 80;
+				double containerCpu = 1;
+				double containerMemoryInGB = 1.5;
+
+				// Add a containergroup with a container
+				ContainerGroup createdContainerGroup = containerClient.ContainerGroups.CreateOrUpdate(resourceGroup.Name, containerGroupName, new ContainerGroup
+				{
+					Location = resourceGroup.Location,
+					OsType = containerOsType,
+					Containers = new List<Container>
+					{
+						new Container
+						{
+							Name = containerName,
+							Image = containerImage,
+							Ports = new List<ContainerPort>
+							{
+								new ContainerPort(containerPort)
+							},
+							Resources = new ResourceRequirements
+							{
+								Requests = new ResourceRequests
+								{
+									Cpu = containerCpu,
+									MemoryInGB = containerMemoryInGB
+								}
+							}
+						}
+					}
+				});
+				Assert.NotNull(createdContainerGroup);
+				Assert.Equal(containerGroupName, createdContainerGroup.Name);
+				Assert.Equal(containerOsType, createdContainerGroup.OsType);
+				Assert.Equal(1, createdContainerGroup.Containers.Count);
+
+				Container createdContainer = createdContainerGroup.Containers.First();
+				Assert.Equal(containerName, createdContainer.Name);
+				Assert.Equal(containerImage, createdContainer.Image);
+				Assert.Equal(1, createdContainer.Ports.Count);
+				Assert.Equal(containerPort, createdContainer.Ports.First().Port);
+				Assert.Equal(containerCpu, createdContainer.Resources.Requests.Cpu);
+				Assert.Equal(containerMemoryInGB, createdContainer.Resources.Requests.MemoryInGB);
+
+				// Wait till the container group is ready before proceeding
+				ContainerGroupUtilities.WaitTillProvisioningStateSucceeded(containerClient, resourceGroup.Name, containerGroupName);
```

Comment:
Not really necessary. You can call get or list before reach succeeded provisioning state.